ExampleApp/ExampleApp/Screens/Characteristics/ViewController/CharacterisiticsViewController.swift

```diff
@@ -0,0 +1,175 @@
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setDataSourceRefreshBlock()
+        bindViewModel()
+    }
+
+    private func setupTableView() {
+        customView.setTableView(dataSource: dataSource, delegate: self)
+        registerCells()
+    }
+
+    private func setDataSourceRefreshBlock() {
+        self.dataSource.refreshDataBlock = { [weak self] in
+            self?.customView.refreshTableView()
+        }
+    }
+
+    private func bindViewModel() {
+        subscribeViewModelOutputs()
+        dataSource.bindItemsObserver(to: viewModel.characteristicsOutput)
+    }
+
+    private func subscribeViewModelOutputs() {
+        subscribeDataUpdateOutput()
+        subscribeAlertTriggerOutput()
+    }
+
+    private func subscribeDataUpdateOutput() {
+        viewModel.dataUpdateOutput.subscribe(onNext: { [unowned self]  _ in
+            self.customView.refreshTableView()
+        }).disposed(by: disposeBag)
+    }
+
+    private func subscribeAlertTriggerOutput() {
+        viewModel.alertTriggerOutput.subscribe(onNext: { [unowned self] message in
+            self.showAlert(title: Constant.Strings.titleSuccess, message: message)
+        }).disposed(by: disposeBag)
+    }
+
+    private func registerCells() {
+        customView.register(cellType: CharacterisiticsTableViewCell.self,
+                forCellReuseIdentifier: String(describing: CharacterisiticsTableViewCell.self))
+    }
+
+    private func addWriteActions(to actionSheet: UIAlertController) {
+        let writeValueNotificationAction = UIAlertAction(title: Constant.Strings.titleWrite,
+                style: .default) { _ in
+            self.showWriteFieldForCharacteristic()
+        }
+        actionSheet.addAction(writeValueNotificationAction)
+    }
+
+    private func addReadActions(to actionSheet: UIAlertController) {
+        let readValueNotificationAction = UIAlertAction(title: Constant.Strings.titleRead,
+                style: .default) { [weak self] _ in
+            self?.viewModel.triggerValueRead()
+        }
+        actionSheet.addAction(readValueNotificationAction)
+    }
+
+    private func addNotificationActions(to actionSheet: UIAlertController) {
+        let turnNotificationOffAction = UIAlertAction(title: Constant.Strings.turnOffNotifications,
+                style: .default) { _ in
+            self.viewModel.setNotificationsState(enabled: false)
+        }
+        let turnNotificationOnAction = UIAlertAction(title: Constant.Strings.turnOnNotifications,
+                style: .default) { _ in
+            self.viewModel.setNotificationsState(enabled: true)
+        }
+        actionSheet.addAction(turnNotificationOffAction)
+        actionSheet.addAction(turnNotificationOnAction)
+    }
+
+    private func addDismissAction(to actionSheet: UIAlertController) {
+        let dismissAction = UIAlertAction(title: Constant.Strings.titleCancel,
+                style: .cancel) { [unowned self] _ in
```

Comment:
no need for `unowned` since UIAlertAction is inside function

Reply:
Done